File: adf_core_python/core/component/communication/communication_module.py
from abc import ABC, abstractmethod

from rcrs_core.agents.agent import Agent

from adf_core_python.core.agent.communication.message_manager import MessageManager


class CommunicationModule(ABC):
    @abstractmethod
    def receive(self, agent: Agent, message_manager: MessageManager) -> None:
        pass

    @abstractmethod
    def send(self, agent: Agent, message_manager: MessageManager) -> None:
        pass

File: java/lib/src/main/java/adf_core_python/component/communication/CommunicationModule.java
package adf_core_python.component.communication;

import adf_core_python.agent.Agent;
import adf_core_python.agent.communication.MessageManager;

abstract public class CommunicationModule {

    abstract public void receive(Agent agent, MessageManager messageManager);

    abstract public void send(Agent agent, MessageManager messageManager);
}
